sandbox/plc4c/spi/include/plc4c/spi/read_buffer.h

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+#ifndef PLC4C_READ_BUFFER_H_
+#define PLC4C_READ_BUFFER_H_
+
+#include <stdint.h>
+#include <plc4c/system.h>
+#include <plc4c/types.h>
+
+struct plc4c_spi_read_buffer {
+
+};
+typedef struct plc4c_spi_read_buffer plc4c_spi_read_buffer;
+
+uint32_t plc4c_spi_read_get_pos(plc4c_spi_read_buffer* buf);
+
+uint32_t plc4c_spi_read_get_total_bytes(plc4c_spi_read_buffer* buf);
+
```

Comment:
bytes not bits?

Reply:
Well these are not methods used for the normal parsing. All of the normal read methods are bit-based

Reply:
Can you point me to an example reading bits?
Why would would you pass in num_bits to read a field of a certain size like float?  Are you saying that not only is the buffer abstracting reading types from a byte array, but is also abstracting reading types from a byte array with some packing scheme?

Maybe you should put something in the headers about what you are going after?

I think there should be methods for reading just a byte* of n bytes out as well

Reply:
in this case you would have an "array" typed field of type "byte" ...

Reply:
I'll wait to look at this until it is commented, so I can understand what you are doing.